products/sle/main.pm

```diff
@@ -278,7 +278,9 @@ if (sle_version_at_least('15') && !check_var('SCC_REGISTER', 'installation')) {
         my $addonurl;
 
         for my $short_name (@modules) {
-            my $full_name = $registration::SLE15_MODULES{$short_name};
+            # Map the module's full name from its short name.
+            # If it's not defined in $registration::SLE15_MODULES, then assume it's a Product/Extension and treat accordingly
+            my $full_name = $registration::SLE15_MODULES{$short_name} ? $registration::SLE15_MODULES{$short_name} : uc $short_name;
```

Comment:
What about creating hash for SLE15_PRODUCTS and put products there, and then we know if the key is from that hash it's a product and another repo name should be used. Then we can have some flag like is_product or is_module. I'm just afraid that this code will explode =)

Reply:
Thought about it, but currently all products are already using short names but in all caps (Product-SLES, Product-SLED, Product-SLES_SAP, Product-HPC and Product_WE) so decided against it. I agree it could be useful in the long run, but for now the map would be like `{ 'sles' => 'SLES', 'ha' => 'HA' }`, etc., so it seemed redundant to me.

Reply:
Good point. But what's about having flag? We check if it's in hash, if not it means it's product. Then we can use it in L#296 instead of another string comparison. WDYT?

Reply:
@rwx788 regarding is_product/is_module, you're thinking about having the methods in lib/registration.pm along with SLE15_MODULES and the tentative SLE15_PRODUCTS?

Reply:
I guess as of now it's fine to have it here in the method. If you think it can be useful somewhere else, we could indeed move it to registration module, that's up to you.

Reply:
@rwx788 I went with an `is_module` method in lib/registration.pm next to `%SLE15_MODULES` and `%SLE15_DEFAULT_MODULES` (makes sense to have all related module stuff in the same place, right?), and changes in L#283 and L#296 in main.pm to use the method. Currently doing some test runs, once they complete will push.